### PR TITLE
FEV-355 debounce, print and downloader methods

### PR DESCRIPTION
**Pull request by semarche-kaltura**
added:
- debounce method
- print method
- downloader method
- set ID as not mandatory for cuepointEngine
- add EventType for ui configuration of KalturaPlayer

### CODE REVIEW

File: packages/ui/src/printer.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export function printer(content: string): void {
+    const myWindow = window.open("", "", "width=400,height=600");
+    if (myWindow) {
+        myWindow.document.write(content);
+        myWindow.document.close();
+        myWindow.focus();
+        myWindow.print();
+        myWindow.close();
+    }
```
Comment (eitanavgil) [inline packages/ui/src/printer.ts]: let's make sure we do not leave the myWindow in the memory.

File: packages/ui/src/downloader.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export function downloader(content: string, name: string): void {
```
Comment (esakal) [inline packages/ui/src/downloader.ts]: this is relevant to local content right? if so the name should reflect it. something like `local-content-downloader.ts`

File: packages/ui/src/printer.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export function printer(content: string): void {
```
Comment (esakal) [inline packages/ui/src/printer.ts]: same here, `local-content-printer.ts`.

you can consider putting both the localContentDownloader and localContentPrinter under same file `local-content-utils.ts`

**Comment by esakal** [comment]
@semarche-kaltura feel free to merge conflicts and then squash this PR

**Pull request merged**